language/move-vm/runtime/src/move_vm.rs

```diff
@@ -38,4 +39,8 @@ impl MoveVM {
     pub fn new_session<'r, S: MoveStorage>(&self, remote: &'r S) -> Session<'r, '_, S> {
         self.runtime.new_session(remote)
     }
+
+    pub fn loader(&self) -> &Loader {
```

Comment:
It seems a bit odd to expose this by asking clients to create a VM and then asking clients to use the loader via the VM even if they have no desire to execute code--maybe a convenient way to construct a `Loader` outside of the VM would make more sense?

Reply:
With that being said, I also find it weird to expose the loader directly. For now, I'd suggest that we expose a few APIs via the Session instead:
```
impl Session {
    pub fn get_type_layout(
        &self,
        type_tag: &TypeTag,
        move_storage: &impl MoveStorage,
    ) -> VMResult<MoveTypeLayout>;

    pub fn get_function_signature(
        &self,
        function_name: &IdentStr,
        module_id: &ModuleId,
        ty_args: &[TypeTag],
        move_storage: &impl MoveStorage,
    ) -> VMResult<(Vec<TypeTag>, Vec<TypeTag>)>;
}
```
This way we can also avoid creating that dummy transaction data cache.

Reply:
Sorry for taking so long to reply!

I very much like the long-term vision. I am less bothered about exposing the loader if we're going to fix it soon (TM), but the session-based API's would also be ok by me.